### PR TITLE
Kinds: publish kinds to kind registry

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What is this feature?**
This PR adds three GH workflows:
- `verify-kinds` verifies that latest kinds do not contain updates on schemas that were already published: runs on every PR
- `publish-kinds-next` verifies the above + publishes kinds to the remote [kind registry](https://github.com/grafana/kind-registry) in the `next` folder: runs on every commit to main
- `publish-kinds-next` verifies the above + publishes kinds to the remote [kind registry](https://github.com/grafana/kind-registry) in the `<release version>` folder: runs on every release

**Why do we need this feature?**
This will guarantee the immutability of the published schemas, hence this is a part of unlocking the stable maturity level for schemas.

**Who is this feature for?**
This is for CUE kinds authors (and the registry will also be used in as-code tooling).

**Special notes for your reviewer:**
What should be discussed before merging (or could be iterated on after merging):
- Schemas organization in the registry
- Ways of loading kinds when comparing published and current kinds
- Formatting of published schemas (we should test that as-code tooling can use the published schemas as they currently use Grafana schemas - with no unwanted changes to their generated outputs).

What remains:
- [x] Clean up the GH workflows and make the publishing one work for releases
- [x] Add a way to run the `verify-kinds` script for one kind only to run it locally in case of errors
- [x] ~~Create the GH token `PUBLISH_TO_KIND_REGISTRY_TOKEN`~~ Add step to generate token

~~Depends on: https://github.com/grafana/thema/pull/130~~ (merged)

### CODE REVIEW

**Review by joanlopez — COMMENTED** [review]
Haven't had time to carefully look at all the details, but so far I left some minor comments.
In any case, it looks really promising 🔥 , greaaat! 🚀

File: .github/workflows/scripts/kinds/verify-kinds.go [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+	// Kind verification
+	for _, kind := range corekind.NewBase(nil).All() {
+		if len(kindArgs) > 0 && !contains(kindArgs, kind.Name()) {
+			continue
+		}
+
+		name := kind.Props().Common().MachineName
+		err := verifyKind(kind, name, "core", latestRegistryDir)
+		if err != nil {
+			errs = append(errs, err)
```
[the remark this answers is not in the document]
Comment (AgnesToulet) [reply .github/workflows/scripts/kinds/verify-kinds.go:80]: Good idea!

File: .github/workflows/scripts/kinds/verify-kinds.go [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+		}
+
+		for _, kind := range pp.ComposableKinds {
+			if len(kindArgs) > 0 && !contains(kindArgs, kind.Name()) {
+				continue
+			}
+
+			si, err := kindsys.FindSchemaInterface(kind.Def().Properties.SchemaInterface)
+			if err != nil {
+				die(err)
```
[the remark this answers is not in the document]
Comment (AgnesToulet) [reply .github/workflows/scripts/kinds/verify-kinds.go:100]: I think the idea was that this should never fail. If this is failing, you are not in a state where you want to publish schemas as your schema is not even valid. But it doesn't hurt to not exit and show the other errors as well, I'll make the change!

File: .github/workflows/scripts/kinds/verify-kinds.go [hunk 3]
```diff
@@ -0,0 +1,415 @@
...
+	}
+}
+
+func die(err error) {
+	if err != nil {
+		fmt.Fprint(os.Stderr, err, "\n")
+		fmt.Println("Run `go run verify-kinds.go <kind name> <kind name>` to run the script on a limited set of kinds.")
+		os.Exit(1)
+	}
+}
```
[the remark this answers is not in the document]
Comment (AgnesToulet) [reply .github/workflows/scripts/kinds/verify-kinds.go:156]: Great idea, thanks! I didn't think of the `...error`.

File: .github/workflows/scripts/kinds/verify-kinds.go [hunk 4]
```diff
@@ -0,0 +1,414 @@
...
+
+	// File generation
+	jfs := codejen.NewFS()
+	registryPath := filepath.Join(".github", "workflows", "scripts", "kinds")
+
+	coreJennies := codejen.JennyList[kindsys.Kind]{}
+	coreJennies.Append(
+		KindRegistryJenny(registryPath, kindArgs),
+	)
+	corefs, err := coreJennies.GenerateFS(corek...)
```
Comment (sdboyer) [inline .github/workflows/scripts/kinds/verify-kinds.go:130]: More of an FYI note - if there's just one jenny actually going to be used here (the `KindRegistryJenny`), it's not actually necessary to create a `JennyList`.

It's fine, even preferable to keep this as-is given that we do it everywhere else, and until we make some improvements to the jenny library it's less familiar to do things that way

File: pkg/cuectx/load.go [hunk 5]
```diff
@@ -136,6 +136,9 @@ func LoadGrafanaInstance(relpath string, pkg string, overlay fs.FS) (*build.Inst
 		if err != nil {
 			return nil, err
 		}
+
+		// merge the kindsys filesystem with ours for the kind categories
+		f = merged_fs.NewMergedFS(kindsys.CueSchemaFS, f)
```
Comment (sdboyer) [inline pkg/cuectx/load.go:141]: whoops! good catch

File: pkg/plugins/pfs/pfs.go [hunk 6]
```diff
@@ -232,6 +232,41 @@ func ParsePluginFS(fsys fs.FS, rt *thema.Runtime) (ParsedPlugin, error) {
 	return pp, nil
 }
 
+func LoadComposableKindDef(fsys fs.FS, rt *thema.Runtime, defpath string) (kindsys.Def[kindsys.ComposableProperties], error) {
```
Comment (sdboyer) [inline pkg/plugins/pfs/pfs.go:235]: It's pretty significant to expose a new one of these loaders. This'll definitely need to move as we move pfs around more generally.

Seems OK for now, though.

Would prefer to see a comment on it!

**Review by sdboyer — APPROVED** [review]
LGTM!

**Pull request merged**